concepts/arrays/about.md

```diff
@@ -1 +1,158 @@
+
+```
+fn main() {
+    // Create an array to hold unsigned 128-bit integers
+    let mut age = ArrayTrait::<u128>::new();
+
+    // Add an element to the array
+    age.append(18);
+
+    // Define the index to access (change this to see different results)
+    let index_to_access = 1;
+
+    // Use match to handle the result of accessing the element at the specified index
+    match arr.get(index_to_access) {
+        Option::Some(value) => {
+            // Print the value if the element exists at the index
+            println!("Element at index {}: {}", index_to_access, *value);
+        },
+        Option::None => {
+            // Panic with a message if the index is out of bounds
+            panic!("Index {} is out of bounds!", index_to_access);
+        }
+    }
+}
+
+```
+
+#### 2. Using at() Method:
+
+The at() method directly gives you the value of an element at a specific position in an array-like structure. It uses an unbox() operation to extract the stored value from a container. If you try to access an index that doesn't exist (out-of-bounds), the program will crash with an error. This method is useful when you want your program to stop if it encounters an unexpected situation like accessing an invalid index.
+
+Here's an example:
+
+```markdown
+fn main() {
+// Assume we have an array-like structure called `ArrayTrait`
+let mut a = ArrayTrait::new();
+
+    // Add some elements to the array
+    a.append(0);
+    a.append(1);
+
+    // Access elements using `at()` method
+    let first = *a.at(0);  // Access element at index 0
+    let second = *a.at(1); // Access element at index 1
+
+    // Now `first` will be 0 and `second` will be 1
+
+}
+```
+
+# Removing Array Elements
+
+The only way to remove elements from an array in Cairo is from the front. This operation is facilitated by the pop_front() method, which removes and returns the first element of the array.
+
+Let's walk through an example to illustrate how this works:
+
+```
+fn main() {
+    // Assume we have an array-like structure called `ArrayTrait`
+    let mut a = ArrayTrait::new();
+
+    // Add some elements to the array
+    a.append(10);
+    a.append(1);
+    a.append(2);
+
+    // Remove the first element from the front of the array
+    let first_value = a.pop_front().unwrap(); // `unwrap()` is used to get the value from `Option`
+
+    // Print the removed value
+    println!("The first value is {}", first_value);
+}
+
+```
+
+The `pop_front()` method is used to remove the first element from the front of the array. In this case, 10 is removed.
+
+`pop_front()` returns an Option. This allows for handling situations where the array might be empty (though not shown explicitly here).
+
+`unwrap()` is used here assuming the array isn't empty, directly retrieving the value of the removed element (10). In real-world scenarios, error handling would typically be more robust.
```

Comment:
A couple of additional sections are necessary:
- add a short description of size-related methods
- add a section about the `array!` macro
- describe `Span` struct